File: Project Euler/Largest Prime Factor/pedro_chagas_5485.c
#include <stdio.h>

int solve(long n){
	int i = 2;
	while (i * i <= n) {
		if (n % i != 0) {
			i++;
		}else {
			n = n/i;
		}
	}
	return n;
}

int main(){
	printf("%d\n", solve(600851475143));
}

File: Project Euler/Largest Prime Factor/pedro_chagas_5485.rs
fn solve(n: u64) -> u64{
    let mut i: u64 = 2;
    let mut n_mutable: u64 = n;
    while i * i <= n_mutable {
	if n_mutable % i != 0 {
	    i += 1;
	}else{
	    n_mutable = n_mutable/i;
	}
    }
    return n_mutable;

}

fn main(){
    let solution: u64 = solve(600851475143);
    println!("{}", solution);
}
